### ISSUE TITLE
Fixing: "Failed to get 'code.path', skipping this submodule" - related to issue  #6138

### ISSUE THREAD

**Issue by TimoLassmann** [datalad/datalad#6334]
### What is the problem?

I am having exactly the same problem as described here:

"datalad clone does not actually stop when target path not empty #6138."

Below is just a copy of the previous issue that I believe was fixed.

My question is how to properly  fix a dataset with this error.

datalad clone should do nothing if target is not empty, but it currently proceeds with some steps and corrupts the dataset. Any modified file in the target path gets saved, and you get [WARNING] Failed to get 'code.path', skipping this submodule whenever you do anything with subdatasets.






### What steps will reproduce the problem?

Have a non-empty directory dir in dataset
datalad clone -d^ $URL dir (very plausible user error)



### Datalad information

datalad 0.15.2

## system 
  - distribution: arch
  - encoding: 
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - max_path_length: 270
  - name: Linux
  - release: 5.10.75-1-lts
  - type: posix
  - version: #1 SMP Wed, 20 Oct 2021 11:02:09 +0000


### Additional context


Example log

install(error): code (dataset) [target path already exists and not empty, refuse to clone into target path]
add(ok): code/ba.sh (file)                                                                                                                                                                                                                    
save(ok): . (dataset)                                                                                                                                                                                                                         
[WARNING] Failed to get 'code.path', skipping this submodule                                                                                                                                                                                  
[WARNING] Failed to get 'code.path', skipping this submodule 
add(ok): .gitmodules (file)                                                                                                                                                                                                                   
save(ok): . (dataset)
action summary:
  add (ok: 2)
  install (error: 1)
  save (ok: 2)

### Have you had any success using DataLad before?

Yep

**Comment by adswa** [comment]
Hey, thanks for getting in touch!

Indeed, there is a fix for the problem you're describing, but it is released in version ``0.15.4``. The question on how to fix such a dataset properly depends a bit on what is broken. If its the ``[WARNING] Failed to get 'funky.path', skipping this submodule`` only, you could either remove the partially created subdataset entry within the ``.gitmodules`` file of the superdataset, or, if you actually proceeded to properly install a subdataset under this path, patch the incomplete record  by the missing key value pairs by hand. In both cases, you would need to find an entry ``[submodule <non-empty-path"]`` in ``.gitmodules`` of the superdataset, remove it completely or patch it, and save the file by hand.
```
❱ cat .gitmodules 
[submodule "exists"]                                                               <-- normal
	path = exists
	url = https://gin.g-node.org/adswa/bids-data
	datalad-id = 68bdb3f3-eafa-4a48-bddd-31e94e8b8242
	datalad-url = https://gin.g-node.org/adswa/bids-data
[submodule "funky"]                                                              <- needs attention
	datalad-url = /data/project/enki/processed/code/pipelines
```

Or is the problem trickier, e.g., were some files accidentally lost, or do you want to remove the erroneous commit?

**Comment by TimoLassmann** [comment]
Hi, this did the trick. thank you!
